src/resolvers/user.ts

```diff
@@ -1,36 +1,322 @@
+    },
+
+    // Deactivate the specified user Account (SuperAdmin and Admin only)
+    deactivateUser: async (_, { userId }, context: MyContext): Promise<User> => {
+      const ref = 'deactivateUser resolver';
+      if (isAdmin(context.token)) {
+        const result = await User.findById(ref, context, userId);
+        // For some reason these are being returned a Objects and not User!
+        const user = new User(result);
+        // Only continue if the current user's affiliation matches the user OR they are SuperAdmin
+        if (context.token.affiliationId === user.affiliationId || isSuperAdmin(context.token)) {
+          user.active = false;
+          const updated = await User.update(context, new User(user).tableName, user, ref, ['password']);
+
+          if (!updated) {
+            throw InternalServerError('Unable to deactivate the user at this time');
+          }
+          // Return the result, because updated will not return a User since they are now inactive
+          return result as User;
+        }
+      }
+      // Unauthorized!
+      throw context?.token ? ForbiddenError() : AuthenticationError();
+    },
+    // Reactivate the specified user Account (SuperAdmin and Admin only)
+    activateUser: async (_, { userId }, context: MyContext): Promise<User> => {
+      const ref = 'activateUser resolver';
+      if (isAdmin(context.token)) {
+        const result = await User.findById(ref, context, userId);
+        // For some reason these are being returned a Objects and not User!
+        const user = new User(result);
+        // Only continue if the current user's affiliation matches the user OR they are SuperAdmin
+        if (context.token.affiliationId === user.affiliationId || isSuperAdmin(context.token)) {
+          user.active = true;
+          const updated = await User.update(context, user.tableName, user, ref, ['password']);
+
+          if (!updated) {
+            throw InternalServerError('Unable to activate the user at this time');
+          }
+          return result as User;
+        }
+      }
+      // Unauthorized!
+      throw context?.token ? ForbiddenError() : AuthenticationError();
+    },
+    // Merge the 2 user accounts (SuperAdmin and Admin only)
+    mergeUsers: async (_, { userIdToBeMerged, userIdToKeep }, context: MyContext): Promise<User> => {
+      if (isAdmin(context.token)) {
+        const userToMerge = await User.findById('mergeUsers resolver', context, userIdToBeMerged);
+        const userToKeep = await User.findById('mergeUsers resolver', context, userIdToKeep);
+        // Only continue if the current user's affiliation matches the user OR they are SuperAdmin
+        const affil = context.token.affiliationId;
+        if (
+          (affil === userToMerge.affiliationId && affil === userToKeep.affiliationId) ||
+          isSuperAdmin(context.token)
+        ) {
+          const merged = await mergeUsers(context, userToMerge, userToKeep);
```

Comment:
L300: When you pass `userToKeep` into `mergeUsers`, I think you have to create an instance of the "User" class for that, because when I try and merge two users, I get an error that "**userToKeep.update is not a function**", which is reminiscent of the other cases where I had to create a new instance of a class in order to access its methods.

Reply:
I am wondering why this happens and think maybe the issue is that the `MySQLModel.query` returns its result(s) which contains all of the properties but not the functions that Model class has, so I think I was getting confused because `user.email` works but `user.update` doesn't.

I will add a note to that new cleanup ticket to update this. It will be good to update all of those functions so that we return an instance of the model instead of just the query result

Reply:
Yea, it would be good to create a ticket to investigate that.

Reply:
Maybe make a note here or in #137 that `mergeUsers` needs to be fixed due to this error, just so we don't forget that this function isn't working and needs to be updated.

Reply:
Oh, I thought I had fixed the issue. I'll add a note to the ticket

Reply:
Ok thanks Brian.